src/geometry/CTiglBSplineAlgorithms.h

```diff
@@ -262,6 +292,37 @@ class CTiglBSplineAlgorithms
     TIGL_EXPORT static Handle(Geom_BSplineCurve) concatCurves(std::vector<Handle(Geom_BSplineCurve)> curves,
                                                               bool parByLength=true, double tolerance = 1e-6);
 
+    /**
+     * @brief Concatenates two surfaces in u direction
+     *
+     * Note: This function has the following requirements
+     *   - s2 follows s1 in u direction and have a common boundary
+     *   - s2 follows s1 also in parametric domain
+     *   - both surfaces are non-rational!
+     *   - the surfaces a non-periodic in u direction
+     *
+     * The resulting surface might not C1 and C2 continuous.
+     * We still have geometric continuity.
+     *
+     * To achieve parametric continuity, the surfaces must be
+     * re-parametrized accordingly before
+     */
+    TIGL_EXPORT static Handle(Geom_BSplineSurface) concatSurfacesUDir(Handle(Geom_BSplineSurface) bspl1, Handle(Geom_BSplineSurface) bspl2, double space_tol=1e-5);
+
+    /**
+     * @ brief Approximates a surface by sampling it and interpolating it
+     * with a degree 3 b-spline surface
+     *
+     * The approximation error depends on the number of segments.
+     * The more segments are used, the better will be the result.
+     *
+     * Currently, derivatives at the boundary are not taken into account.
+     *
+     * @param surf Surface to approximate
+     * @param nseg_u Number of knot segments in u direction
+     * @param nseg_v Number of knot segments in v direction
+    */
+    TIGL_EXPORT static Handle(Geom_BSplineSurface) approxSurface(Handle(Geom_BSplineSurface) surf, unsigned int nseg_u, unsigned int nseg_v);
```

Comment:
You are right, the name is not good.

The intention of the function is to recreate the function with another, equidistant knot vector. Ideally, the function would also take the surface boundary derivatives into account, but I was too lazy yet to implement it.

Technically, it is currently a interpolation, but it could be anything else actually. So, I'd like to address more the intention than the actual implementation detail in the name.

Reply:
Here are some ideas for a better name:
 - improveKnotDistribution(surf...)
 - makeKnotsUniform(surf...)

Reply:
I like the second suggestion! The first one implies that uniform is always better. This might not be true in all cases.